model_compression_toolkit/core/common/quantization/bit_width_config.py

```diff
@@ -32,6 +33,64 @@ class ManualBitWidthSelection:
     filter: BaseNodeMatcher
     bit_width: int
 
+@dataclass
+class ManualWeightsBitWidthSelection(ManualBitWidthSelection):
+    """
+   Class to encapsulate the manual weights bit width selection configuration for a specific filter.
+
+   Attributes:
+       filter (BaseNodeMatcher): The filter used to select nodes for bit width manipulation.
+       attr (str): The attribute used to select nodes for bit width manipulation. [KERNEL_ATTR|BIAS_ATTR]
+   """
+    attr: str
+
+def _expand_to_list_core(
+        filters: Union[List[BaseNodeMatcher]],
+        vals: Union[List[any], any]):
+    vals = [vals] if not isinstance(vals, list) else vals
+    if len(vals) > 1 and len(vals) != len(filters):
+        Logger.critical(f"Configuration Error: The number of provided bit_width values {len(vals)} "
+                        f"must match the number of filters {len(filters)}, or a single bit_width value "
+                        f"should be provided for all filters.")
+    elif len(vals) == 1 and len(filters) > 1:
+        vals = [vals[0] for f in filters]
+    return vals
+
+def _expand_to_list_filter_and_bit_width(
+        filters: Union[List[BaseNodeMatcher]],
+        bit_widths: Union[List[int], int],
+        attrs: Union[List[str], str] = None):
+    filters = [filters] if not isinstance(filters, list) else filters
+    bit_widths = _expand_to_list_core(filters, bit_widths)
+    attrs = _expand_to_list_core(filters, attrs)
+
+    return attrs, bit_widths, filters
+
+def _make_nodes_to_change_bit_width(graph, manual_bit_width_selection_list):
+    unit_nodes_to_change_bit_width = {}
+    for manual_bit_width_selection in manual_bit_width_selection_list:
+        filtered_nodes = graph.filter(manual_bit_width_selection.filter)
+        if len(filtered_nodes) == 0:
+            Logger.critical(
+                f"Node Filtering Error: No nodes found in the graph for filter {manual_bit_width_selection.filter.__dict__} "
+                f"to change their bit width to {manual_bit_width_selection.bit_width}.")
+        for n in filtered_nodes:
+            # check if a manual configuration exists for this node
+            if n in unit_nodes_to_change_bit_width:
+                Logger.info(
+                    f"Node {n} has an existing manual bit width configuration of {unit_nodes_to_change_bit_width.get(n)}. A new manual configuration request of {manual_bit_width_selection.bit_width} has been received, and the previous value is being overridden.")
+            if isinstance(manual_bit_width_selection_list,  ManualBitWidthSelection):
```

Comment:
Not sure that this would work, since any object of `ManualWeightsBitWidthSelection` is also an instance of `ManualBitWidthSelection`. It would work if you switch the order of the if statements.

However, regarding my previous comment, if you split this function - one for activations and one for weights you won't be needing this `isinstance` separation